src/Materials/babylon.materialHelper.ts

```diff
@@ -111,11 +111,11 @@
                     defines["DIRLIGHT" + lightIndex] = false;
 
                     var type;
-                    if (light instanceof SpotLight) {
+                    if (light.getClassName() === "SpotLight") {

```

Comment:
for light tests, I now use getTypeID instead.  It already exists & is an integer.  Think I need to close this PR and open a new one to add another commit.